pkgs/development/cuda-modules/flags.nix

```diff
@@ -0,0 +1,318 @@
+
+  formatCapabilities = {
+    cudaCapabilities,
+    enableForwardCompat ? true,
+  }: rec {
+    inherit cudaCapabilities enableForwardCompat;
+
+    # archNames :: List String
+    # E.g. [ "Turing" "Ampere" ]
+    archNames = lists.unique (builtins.map (cap: cudaComputeCapabilityToName.${cap}) cudaCapabilities);
+
+    # realArches :: List String
+    # The real architectures are physical architectures supported by the CUDA version.
+    # E.g. [ "sm_75" "sm_86" ]
+    realArches = archMapper "sm" cudaCapabilities;
+
+    # virtualArches :: List String
+    # The virtual architectures are typically used for forward compatibility, when trying to support
+    # an architecture newer than the CUDA version allows.
+    # E.g. [ "compute_75" "compute_86" ]
+    virtualArches = archMapper "compute" cudaCapabilities;
+
+    # arches :: List String
+    # By default, build for all supported architectures and forward compatibility via a virtual
+    # architecture for the newest supported architecture.
+    # E.g. [ "sm_75" "sm_86" "compute_86" ]
+    arches =
+      realArches
+      ++ lists.optional enableForwardCompat (lists.last virtualArches);
+
+    # gencode :: List String
+    # A list of CUDA gencode arguments to pass to NVCC.
+    # E.g. [ "-gencode=arch=compute_75,code=sm_75" ... "-gencode=arch=compute_86,code=compute_86" ]
+    gencode = let
+      base = gencodeMapper "sm" cudaCapabilities;
+      forward = gencodeMapper "compute" [(lists.last cudaCapabilities)];
+    in
+      base ++ lib.optionals enableForwardCompat forward;
+
+    # gencodeString :: String
+    # A space-separated string of CUDA gencode arguments to pass to NVCC.
+    # E.g. "-gencode=arch=compute_75,code=sm_75 ... -gencode=arch=compute_86,code=compute_86"
+    gencodeString = strings.concatStringsSep " " gencode;
+
+    # Jetson devices cannot be targeted by the same binaries which target non-Jetson devices. While
+    # NVIDIA provides both `linux-aarch64` and `linux-sbsa` packages, which both target `aarch64`,
+    # they are built with different settings and cannot be mixed.
+    # isJetsonBuild :: Boolean
+    isJetsonBuild = let
+      # List of booleans representing whether any of the currently targeted capabilities are
+      # Jetson devices.
+      # isJetsons :: List Boolean
+      isJetsons =
+        lists.map
+        (trivial.flip builtins.getAttr cudaComputeCapabilityToIsJetson)
+        cudaCapabilities;
+      anyJetsons = lists.any (trivial.id) isJetsons;
+      allJetsons = lists.all (trivial.id) isJetsons;
+      hostIsAarch64 = hostPlatform.isAarch64;
+    in
+      trivial.throwIfNot
+      (anyJetsons -> (allJetsons && hostIsAarch64))
+      ''
+        Jetson devices cannot be targeted with non-Jetson devices. Additionally, they require hostPlatform to be aarch64.
+        You requested ${builtins.toJSON cudaCapabilities} for host platform ${hostPlatform.system}.
+        Exactly one of the following must be true:
```

Comment:
```bash
❯ nix eval github:ConnorBaker/nixpkgs/feat/cuda-redist-multiple-arch#legacyPackages.aarch64-linux.cudaPackages.cuda_compat.src
error:
       … in the left operand of the update (//) operator
...

       error: Jetson devices cannot be targeted with non-Jetson devices. Additionally, they require hostPlatform to be aarch64.
       You requested ["6.0","6.1","6.2","7.0","7.2","7.5","8.0","8.6","8.7","8.9","9.0"] for host platform aarch64-linux.
       Exactly one of the following must be true:
       - All CUDA capabilities belong to Jetson devices (false) and the hostPlatform is aarch64 (true).
       - No CUDA capabilities belong to Jetson devices (false).
       See /nix/store/a2gma4qnjx9xg59cq45q8987cx2j0am7-gpus.nix for a list of architectures supported by this version of Nixpkgs.
```

Reply:
Good catch, I indeed still need to use `ALLOW_UNSUPPORTED_PLATFORMS` (or another flags, I don't remember, I set many :sweat_smile: ) and `--impure` to build anything from this PR on the Jetson